# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../kipi-plugins/kipi-plugins-0.2.0-libgpod.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -urpN kipi-plugins/cmake/modules/FindGdk.cmake kipi-plugins-libgpod/cmake/modules/FindGdk.cmake
--- kipi-plugins/cmake/modules/FindGdk.cmake	1970-01-01 07:00:00.000000000 +0700
+++ kipi-plugins-libgpod/cmake/modules/FindGdk.cmake	2009-01-28 17:23:52.000000000 +0600
@@ -0,0 +1,48 @@
+# - Try to find the GDK library
+# Once done this will define
+#
+#  GDK_FOUND - system has GDK
+#  GDK_INCLUDE_DIR - the GDK include directory
+#  GDK_LIBRARIES - Link these to use GDK
+#  GDK_DEFINITIONS - Compiler switches required for using GDK
+#
+
+if (GDK_INCLUDE_DIR AND GDK_LIBRARIES)
+
+  # in cache already
+  SET(GDK_FOUND TRUE)
+
+else (GDK_INCLUDE_DIR AND GDK_LIBRARIES)
+  if(NOT WIN32)
+    # use pkg-config to get the directories and then use these values
+    # in the FIND_PATH() and FIND_LIBRARY() calls
+    INCLUDE(UsePkgConfig)
+  
+    PKGCONFIG(gdk-pixbuf-2.0 _GDKIncDir _GDKLinkDir _GDKLinkFlags _GDKCflags)
+  
+    set(GDK_DEFINITIONS ${_GDKCflags})
+  endif(NOT WIN32)
+
+  FIND_PATH(GDK_INCLUDE_DIR gdk-pixbuf/gdk-pixbuf.h /usr/include/gtk-2.0
+    ${_GDKIncDir}
+  )
+  
+  FIND_LIBRARY(GDK_LIBRARIES NAMES gdk_pixbuf-2.0
+    PATHS
+    ${_GDKLinkDir}
+  )
+
+  if (GDK_INCLUDE_DIR AND GDK_LIBRARIES)
+    SET(GDK_FOUND TRUE)
+  else (GDK_INCLUDE_DIR AND GDK_LIBRARIES)
+    SET(GDK_FOUND FALSE)
+  endif (GDK_INCLUDE_DIR AND GDK_LIBRARIES)
+
+  include(FindPackageHandleStandardArgs)
+  FIND_PACKAGE_HANDLE_STANDARD_ARGS(Gdk DEFAULT_MSG GDK_INCLUDE_DIR GDK_LIBRARIES )
+ 
+  MARK_AS_ADVANCED(GDK_INCLUDE_DIR GDK_LIBRARIES)
+  
+endif (GDK_INCLUDE_DIR AND GDK_LIBRARIES)
+
+macro_bool_to_01(GDK_FOUND GDK_FOUND)
diff -urpN kipi-plugins/cmake/modules/FindGpod.cmake kipi-plugins-libgpod/cmake/modules/FindGpod.cmake
--- kipi-plugins/cmake/modules/FindGpod.cmake	2009-01-28 17:15:15.000000000 +0600
+++ kipi-plugins-libgpod/cmake/modules/FindGpod.cmake	2009-01-28 18:46:12.000000000 +0600
@@ -23,25 +23,32 @@ if( NOT WIN32 )
   pkg_check_modules(PC_GPOD libgpod-1.0)
 
   set(GPOD_DEFINITIONS ${PC_GPOD_CFLAGS_OTHER})
+  
+  exec_program(${PKGCONFIG_EXECUTABLE} ARGS --atleast-version=0.7.0 libgpod-1.0 OUTPUT_VARIABLE _pkgconfigDevNull RETURN_VALUE GPOD_VERSION_OKAY)
 
 endif( NOT WIN32 )
 
-FIND_PATH(GPOD_INCLUDE_DIR NAMES gpod/itdb.h
-  PATHS
-  ${PC_GPOD_INCLUDEDIR}
-  ${PC_GPOD_INCLUDE_DIRS}
-  PATH_SUFFIXES gpod-1.0
-)
-
-FIND_LIBRARY(GPOD_LIBRARIES NAMES gpod
-  PATHS
-  ${PC_GPOD_LIBDIR}
-  ${PC_GPOD_LIBRARY_DIRS}
-)
-
-include(FindPackageHandleStandardArgs)
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(Gpod DEFAULT_MSG GPOD_INCLUDE_DIR GPOD_LIBRARIES )
-
-# show the GPOD_INCLUDE_DIR and GPOD_LIBRARIES variables only in the advanced view
-MARK_AS_ADVANCED(GPOD_INCLUDE_DIR GPOD_LIBRARIES )
+if (GPOD_VERSION_OKAY STREQUAL "0")
 
+  FIND_PATH(GPOD_INCLUDE_DIR NAMES gpod/itdb.h
+    PATHS
+    ${PC_GPOD_INCLUDEDIR}
+    ${PC_GPOD_INCLUDE_DIRS}
+    PATH_SUFFIXES gpod-1.0
+  )
+  
+  FIND_LIBRARY(GPOD_LIBRARIES NAMES gpod
+    PATHS
+    ${PC_GPOD_LIBDIR}
+    ${PC_GPOD_LIBRARY_DIRS}
+  )
+  
+  include(FindPackageHandleStandardArgs)
+  FIND_PACKAGE_HANDLE_STANDARD_ARGS(Gpod DEFAULT_MSG GPOD_INCLUDE_DIR GPOD_LIBRARIES )
+  
+  # show the GPOD_INCLUDE_DIR and GPOD_LIBRARIES variables only in the advanced view
+  MARK_AS_ADVANCED(GPOD_INCLUDE_DIR GPOD_LIBRARIES )
+
+else (GPOD_VERSION_OKAY STREQUAL "0")
+    set(GPOD_FOUND FALSE)
+endif (GPOD_VERSION_OKAY STREQUAL "0")
diff -urpN kipi-plugins/CMakeLists.txt kipi-plugins-libgpod/CMakeLists.txt
--- kipi-plugins/CMakeLists.txt	2009-01-28 17:22:17.000000000 +0600
+++ kipi-plugins-libgpod/CMakeLists.txt	2009-01-28 17:30:46.000000000 +0600
@@ -52,6 +52,7 @@ MACRO_OPTIONAL_FIND_PACKAGE(LibXslt)    
 MACRO_OPTIONAL_FIND_PACKAGE(OpenGL)     # For Slideshow and ImageViewer.
 MACRO_OPTIONAL_FIND_PACKAGE(OpenCV)     # For RemoveRedEyes.
 MACRO_OPTIONAL_FIND_PACKAGE(Gpod)       # For ipodexport.
+MACRO_OPTIONAL_FIND_PACKAGE(Gdk)        # For ipodexport.
 MACRO_OPTIONAL_FIND_PACKAGE(GLIB2)      # For ipodexport.
 MACRO_OPTIONAL_FIND_PACKAGE(GObject)    # For ipodexport.
 MACRO_OPTIONAL_FIND_PACKAGE(KdepimLibs) # For Calendar (libkcal).
@@ -201,6 +202,12 @@ ELSE(GPOD_FOUND)
     MESSAGE(STATUS " libgpod library found............... NO  (optional)")
 ENDIF(GPOD_FOUND)
 
+IF(GDK_FOUND)
+    MESSAGE(STATUS " Gdk library found................... YES (optional)")
+ELSE(GDK_FOUND)
+    MESSAGE(STATUS " Gdk library found................... NO  (optional)")
+ENDIF(GDK_FOUND)
+
 IF(KDEPIMLIBS_FOUND)
     MESSAGE(STATUS " libkdepim library found............. YES (optional)")
 ELSE(KDEPIMLIBS_FOUND)
@@ -280,11 +287,11 @@ ELSE(OPENCV_FOUND)
     MESSAGE(STATUS " RemoveRedEyes will be compiled...... NO  (optional - Look README file for more details about dependencies)")
 ENDIF(OPENCV_FOUND)
 
-IF(GPOD_FOUND AND GLIB2_FOUND AND GOBJECT_FOUND)
+IF(GPOD_FOUND AND GLIB2_FOUND AND GOBJECT_FOUND AND GDK_FOUND)
     MESSAGE(STATUS " IpodExport will be compiled......... YES (optional)")
-ELSE(GPOD_FOUND AND GLIB2_FOUND AND GOBJECT_FOUND)
+ELSE(GPOD_FOUND AND GLIB2_FOUND AND GOBJECT_FOUND AND GDK_FOUND)
     MESSAGE(STATUS " IpodExport will be compiled......... NO  (optional - Look README file for more details about dependencies)")
-ENDIF(GPOD_FOUND AND GLIB2_FOUND AND GOBJECT_FOUND)
+ENDIF(GPOD_FOUND AND GLIB2_FOUND AND GOBJECT_FOUND AND GDK_FOUND)
 
 IF(KDEPIMLIBS_FOUND)
     MESSAGE(STATUS " Calendar will be compiled........... YES (optional)")
@@ -359,9 +366,9 @@ IF(KIPIPLUGINS_CAN_BE_COMPILED)
        ADD_SUBDIRECTORY(calendar)
     ENDIF(KDEPIMLIBS_FOUND)
 
-    IF(GPOD_FOUND AND GLIB2_FOUND AND GOBJECT_FOUND)
+    IF(GPOD_FOUND AND GLIB2_FOUND AND GOBJECT_FOUND AND GDK_FOUND)
        ADD_SUBDIRECTORY(ipodexport)
-    ENDIF(GPOD_FOUND AND GLIB2_FOUND AND GOBJECT_FOUND)
+    ENDIF(GPOD_FOUND AND GLIB2_FOUND AND GOBJECT_FOUND AND GDK_FOUND)
 
     IF(OPENCV_FOUND)
         ADD_SUBDIRECTORY(removeredeyes)
diff -urpN kipi-plugins/ipodexport/CMakeLists.txt kipi-plugins-libgpod/ipodexport/CMakeLists.txt
--- kipi-plugins/ipodexport/CMakeLists.txt	2009-01-28 17:19:14.000000000 +0600
+++ kipi-plugins-libgpod/ipodexport/CMakeLists.txt	2009-01-28 17:49:32.000000000 +0600
@@ -1,6 +1,6 @@
 ADD_DEFINITIONS(-DQT3_SUPPORT -DQT3_SUPPORT_WARNINGS)
 
-INCLUDE_DIRECTORIES(${GPOD_INCLUDE_DIR} ${GLIB2_INCLUDE_DIR})
+INCLUDE_DIRECTORIES(${GPOD_INCLUDE_DIR} ${GLIB2_INCLUDE_DIR} ${GDK_INCLUDE_DIR})
 
 SET(kipiplugin_ipodexport_PART_SRCS
     plugin_ipodexport.cpp
@@ -18,6 +18,7 @@ TARGET_LINK_LIBRARIES(kipiplugin_ipodexp
                       ${KDE4_KDE3SUPPORT_LIBS}
                       ${KDE4_KDEUI_LIBS}
                       ${GPOD_LIBRARIES}
+                      ${GDK_LIBRARIES}
                       kipiplugins
                       kipi
                       kio
diff -urpN kipi-plugins/ipodexport/IpodExportDialog.cpp kipi-plugins-libgpod/ipodexport/IpodExportDialog.cpp
--- kipi-plugins/ipodexport/IpodExportDialog.cpp	2009-01-28 17:19:14.000000000 +0600
+++ kipi-plugins-libgpod/ipodexport/IpodExportDialog.cpp	2009-01-28 17:41:08.000000000 +0600
@@ -20,6 +20,11 @@
  *
  * ============================================================ */
 
+// System Includes
+extern "C" {
+#include <gdk-pixbuf/gdk-pixbuf.h>
+}
+
 // Local includes.
 
 #include "IpodHeader.h"
@@ -422,9 +427,13 @@ void UploadDialog::ipodItemSelected( QTr
         return;
 
     Itdb_Artwork *artwork = item->artwork();
-    Itdb_Thumb *thumb = itdb_artwork_get_thumb_by_type( artwork, ITDB_THUMB_PHOTO_SMALL );
+    GdkPixbuf *gpixbuf = NULL;
+    
+    // First arg in itdb_artwork_get_pixbuf(...) is pointer to Itdb_Device struct,
+    // in kipiplugin-ipodexport it is m_itdb->device. i hope it _is_ initialiezed
+    gpixbuf = (GdkPixbuf*) itdb_artwork_get_pixbuf( m_itdb->device, artwork, -1, -1 );
 
-    if( !thumb )
+    if( !gpixbuf )
     {
         kDebug(51000) << "no thumb was found" << endl;
         return;
@@ -439,6 +448,9 @@ void UploadDialog::ipodItemSelected( QTr
 //     QPixmap pix;
 //     pix.convertFromImage( image );
 //     m_ipodPreview->setPixmap( pix );
+    
+    // memory release
+    gdk_pixbuf_unref ( gpixbuf );
 }
 
 void UploadDialog::imageSelected( QTreeWidgetItem *item )
